### ISSUE TITLE
Support for asyncchronous jax-rs?

### ISSUE THREAD

**Issue by Phil-Ba** [vojtechhabarta/typescript-generator#269]
Hello,
I wanted to generate the client code for async rest methods with @Suspended but stumbled across 2 issues:

1. Per definition those methods should not return anything, ie they should be _void_. This however leads the generated client also to have a result type, even though Swagger annotations are present.
2. These methods need an _AsyncResponse_  parameter which will be injected by the container. This parameter is not passed from the client. However it will be generated in the client code.

For issue 1 I found a workaround by specifying _Response_ as return type. This however is a workaround and not really pretty(since it also leads to warnings).
For issue 2 I found no solution yet. I tried adding _AsyncResponse_ to the list of ignored classes, but this will only lead to the client code having a parameter of type any. I looked at _TypeProcessor_ but from what I understand those cant be used to ignore a java type. I tried adding the Swagger @ApiParam(hidden = true) but that didn't work either.

Are there any plans to support this scenario, or any other workarounds I could use?
Example code:
```java
@GET
@ApiOperation(response = Person.class)
public void async(@Suspended AsyncResponse ar) {
        ......
}
```

Expected client code:
`async():RestResponse<Person>;`

**Comment by vojtechhabarta** [comment]
Hello,
unfortunately async JAX-RS is not currently supported and I don't think there is a workaround for issue (2). Maybe solution for (2) will be just that typescript-generator will ignore parameters annotated with `@Suspended` but I will need to look into it.

**Comment by Phil-Ba** [comment]
I think I managed to implement it. I'll post a pull request. Please have a look at it.

**Comment by vojtechhabarta** [comment]
Released [v2.7.447](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.7.447).
